tests/operator/suite.go

```diff
@@ -126,6 +126,46 @@ func LoadChecks() {
 			testOperatorCatalogSourceBundleCount(c, &env)
 			return nil
 		}))
+
+	checksGroup.Add(checksdb.NewCheck(identifiers.GetTestIDAndLabels(identifiers.TestOperatorInstallationInTenantNamespace)).
+		WithSkipCheckFn(testhelper.GetNoOperatorsSkipFn(&env)).
+		WithCheckFn(func(c *checksdb.Check) error {
+			testOperatorInstallationInTenantNamespace(c, &env)
+			return nil
+		}))
+}
+
+// This function checks if SingleNamespaced Operators should only be installed in the tenant dedicated operator namespace
+func testOperatorInstallationInTenantNamespace(check *checksdb.Check, env *provider.TestEnvironment) {
+	check.LogInfo("Starting testInstalledSingleNamespaceOperatorInTenanttNamespace")
+
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+
+	singleNamespacedOperators := filterSingleNamespacedOperatorUnderTest(env.Operators)
+
+	for _, operator := range singleNamespacedOperators {
+		operatorNamespace := operator.Namespace
+
+		check.LogInfo("Checking operator %s in namespace %s ", operator.Name, operator.Namespace)
+
+		doesNamespaceContainsOnlyOperatorPods, err := checkIfNamespaceContainsOnlyOperatorPods(operatorNamespace)
+
+		if err != nil {
+			check.LogError("Skipped - cannot proceed with operator %s", operator.Name)
+			continue
+		}
+		if doesNamespaceContainsOnlyOperatorPods {
+			check.LogInfo("Operator %s has valid installation in tenant namespace %s ", operator.Name, operatorNamespace)
+			compliantObjects = append(compliantObjects, testhelper.NewOperatorReportObject(operator.Namespace, operator.Name,
```

Comment:
Shouldn't the namespace be the compliance object? If a namespace has a mix of pods belonging to operators and not belonging, then the test fail for that namespace. We could also list the pod not belonging to operators under test and are responsible for the test failing and the list of operators belonging to the namespace.